# Практическая работа №3: Обработка выборочных данных. Нахождение интервальных оценок параметров распределения. Проверка статистической гипотезы о нормальном распределении

Выполнили студентки гр. 2383 Щёголева Надежда и Семененко Анна. Вариант №28

## Цель работы

Получение практических навыков вычисления интервальных статистических оценок параметров распределения выборочных данных и проверки «справедливости» статистических гипотез.

## Основные теоретические положения

**Интервальная оценка** - статистическая оценка, которая задается двумя числами (начало и конец интервала где может находится истинное значение параметра $\theta$)


**Точность $\theta^*$** - значение $\delta$ статистическая оценка параметра $\theta^*$, которое удовлетворяет неравенству:

$|\theta^*-\theta|<\delta $

**Наддежность (доверительная вероятность) оценки** $\Theta^*$ - вероятность $\gamma$, для которой выполняется неравенство:

$ \gamma = P\{ \delta > |\Theta - \Theta^*| \} = P\{\Theta^* - \delta < \Theta < \Theta^* + \delta \}$


Доверительный интервал бывает односторонним и двусторонним:

$ \mathbb{P}\{ \beta_1(\gamma) < \Theta < \beta_2(\gamma) \} = \gamma$,

$P\{ \beta(\gamma) < \Theta \} = \gamma$   или   $ P\{ \beta(\gamma) > \Theta \} = \gamma$,

$q = 1-\gamma$ - уровень значимости.

**Доверительный интервал для математического ожидания при неизвестном СКВО**
$$(\bar x_в - t_γS / \sqrt N, \bar x_в + t_γS / \sqrt N)$$



**Нулевая гипотиза $H_0$** - выдвинутая (основная) гипотеза.

**Конкурирующая гипотеза $H_1$** - альтарнативная гипотеза, противоречащая $H_0$.

**Ошибка первого рода** - отвергается верная нулевая гипотеза.

**Ошибка второго рода** - принимается неверная нулевая гипотеза.

**Уровень значимости $\alpha$** - вероятность совешить ошибку первого рода.
**$\beta$** - вероятность ошибки второго рода.

**Мощность критерия $1 - \beta$** - мощность критерия.

Для проверки стат. гипотез используются критерии, с помощью которых вычисляются наблюденные и критические значения случайных величин этих критериев. На основании этих значений выделяют критическую область и область принятия нулевой гипотезы.

## Постановка задачи

Для заданной надежности определить (на основании выборочных данных и результатов выполнения практической работы №2) границы доверительных интервалов для математического ожидания и среднеквадратичного отклонения случайной величины. Проверить гипотезу о нормальном распределении исследуемой случайной величины с помощью критерия Пирсона χ2. Дать содержательную интерпретацию полученным результатам.

## Выполнение работы


In [ ]:
import pandas as pd
import numpy as np
from scipy import stats

In [ ]:
#выборочное срденее, оценка дисперсии и СКО для длины и высоты клюва
x_B_l, D_l, sigma_l, s_l = 43.9, 22.91, 4.79, 4.81
x_B_d, D_d, sigma_d, s_d = 16.14, 4.22, 2.05, 2.06
t_gamma_95 = 1.98
t_gamma_99 = 2.617
q_95 = 0.143
q_99 = 0.198
gamma_values = [0.95, 0.99]
N = 117

df_lenth = pd.read_json('df_stats_len.json')
df_depth = pd.read_json('df_stats_depth.json')

1. Вычислим точность и доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении при заданном объёме выборки для доверительной точности
γ∈{0.95,0.99}.

In [ ]:
def conf_interv_for_M(x_b, s):
  accuracy_95 = t_gamma_95*s/(N)**0.5
  accuracy_99 = t_gamma_99*s/(N)**0.5
  print(accuracy_95)
  print((round(x_b - accuracy_95, 2), round(x_b + accuracy_95, 2)))
  print(accuracy_99)
  print((round(x_b - accuracy_99, 2), round(x_b + accuracy_99, 2)))

In [ ]:
print("Точность и доверительный интервал для длины клюва")
conf_interv_for_M(x_B_l, s_l)

Точность и доверительный интервал для длины клюва
0.880475621468306
(43.02, 44.78)
1.1637397481730085
(42.74, 45.06)


In [ ]:
print("Точность и доверительный интервал для высоты клюва")
conf_interv_for_M(x_B_d, s_d)


Точность и доверительный интервал для высоты клюва
0.3770851933939107
(15.76, 16.52)
0.4983999753090224
(15.64, 16.64)


Длина клюва:

При увеличении доверительной вероятности с 0.95 до 0.99 точность оценки уменьшается.

Поэтому доверительный интервал расширяется:

было: 1.76

стало: 2.32

Высота клюва:

Так же как и для длины клюва, при увеличении доверительной вероятности доверительный интервал расширяется.

Точность оценки уменьшается:

Было: 0.377, стало: 0.498.

Доверительный нтервал становится шире:

было: ширина ≈ 0.76

стало: ширина ≈ 1.00.

2. Для вычисления границ доверительного интервала для среднеквадратичного отклонения определим значение q при заданных γ и n. Построим доверительные интервалы.

In [ ]:
def conf_interv_for_sigma(s):
  print((round(s*(1 - q_95), 3), round(s*(1 + q_95), 3)))
  print((round(s*(1 - q_99), 3), round(s*(1 + q_99), 3)))

In [ ]:
print("Доверительный интервал для СКВО длины клюва")
conf_interv_for_sigma(s_l)

Доверительный интервал для СКВО длины клюва
(4.122, 5.498)
(3.858, 5.762)


In [ ]:
print("Доверительный интервал для СКВО высоты клюва")
conf_interv_for_sigma(s_d)

Доверительный интервал для СКВО высоты клюва
(1.765, 2.355)
(1.652, 2.468)


Для длины клюва при увеличении доверительной вероятности γ с 0.95 до 0.99 интервал расширяется.

Для характеристики высота клюва наблюдается та же закономерность: при увеличении γ доверительный интервал становится шире.

3. Проверим гипотезу о нормальности заданного распределения с помощью критерия $χ^2$ (Пирсона). Для этого найдем теоретические частоты и вычислим наблюдаемое значение критерия.

In [ ]:
# функция для объединения интервалов с абсолютной частотой меньше 5
def merge_intervals(df):
    i = 1
    while i < len(df):
        if df.loc[i, 'Абсолютная частота (m_i)'] < 5:
            prev_interval = df.loc[i-1, 'Интервал']
            curr_interval = df.loc[i, 'Интервал']

            left = min(prev_interval[0], curr_interval[0])
            right = max(prev_interval[-1], curr_interval[-1])

            df.at[i-1, 'Интервал'] = [left, right]
            df.loc[i-1, 'Абсолютная частота (m_i)'] += df.loc[i, 'Абсолютная частота (m_i)']
            df = df.drop(i).reset_index(drop=True)
            i-=1

        else:
            i += 1

    return df

In [ ]:
def table_hi(df, s, x_b):
  data_rows = []
  z_borders = []
  k = int(len(df))
  p_i = np.zeros(k)
  borders = np.array([inter[0] for inter in df['Интервал']] + [df['Интервал'].iloc[-1][1]])

  for i in range(k + 1):
    if i == 0:
        z_borders.append(-np.inf)
    elif i == k:
        z_borders.append(np.inf)
    else:
        z_borders.append((borders[i] - x_b) / s)

  for i in range(k):
    p_i[i] = (stats.norm.cdf(z_borders[i + 1]) - stats.norm.cdf(z_borders[i]))

  n_s = p_i*117
  for i in range(1, k+1):
    inter = df.loc[i]['Интервал']
    n_i = df.loc[i]['Абсолютная частота (m_i)']
    data_rows.append({
        'i': inter,
        'n_i': n_i,
        'p_i': round(p_i[i-1], 2),
        "n'_i": round(n_s[i-1], 3),
        "(n_i-n'_i)²": round((n_i - n_s[i-1])**2, 3),
        "(n_i-n'_i)²/n'_i": round(((n_i - n_s[i-1])**2)/ n_s[i-1], 3),
        "n_i²": n_i**2,
        "n_i²/n'_i": round((n_i**2)/n_s[i-1], 3)})
  df = pd.DataFrame(data_rows, index=range(1, len(data_rows)+1))
  df.loc[len(df) + 1] = [
      'Σ',
      df['n_i'].sum(),
      sum(p_i),
      df["n'_i"].sum(),
      '—',
      df["(n_i-n'_i)²/n'_i"].sum(),
      '—',
      df["n_i²/n'_i"].sum(),
  ]
  display(df)


In [ ]:
df_lenth_merged = merge_intervals(df_lenth)
df_lenth_merged.index = range(1, len(df_lenth_merged) + 1)

In [ ]:
print("Таблица для длины клюва")
table_hi(df_lenth_merged, s_l, x_B_l)

Таблица для длины клюва


,i,n_i,p_i,n'_i,(n_i-n'_i)²,(n_i-n'_i)²/n'_i,n_i²,n_i²/n'_i
1,"[34.5, 37.9]",17,0.11,13.331,13.459,1.010,289,21.678
2,"[38.1, 41.6]",23,0.23,27.200,17.636,0.648,529,19.449
3,"[42.0, 45.2]",23,0.27,31.365,69.979,2.231,529,16.866
4,"[45.3, 48.7]",35,0.24,28.197,46.286,1.642,1225,43.445
5,"[49.0, 59.6]",19,0.14,16.907,4.38,0.259,361,21.352
6,Σ,117,1.00,117.000,—,5.790,—,122.790


In [ ]:
print("Таблица для высоты клюва")
table_hi(df_depth, s_d, x_B_d)

Таблица для высоты клюва


,i,n_i,p_i,n'_i,(n_i-n'_i)²,(n_i-n'_i)²/n'_i,n_i²,n_i²/n'_i
1,"[13.1, 14.2]",23,0.19,21.747,1.569,0.072,529,24.325
2,"[14.3, 15.3]",31,0.17,20.339,113.651,5.588,961,47.248
3,"[15.4, 16.5]",19,0.23,26.750,60.068,2.245,361,13.495
4,"[16.6, 17.5]",11,0.19,21.904,118.888,5.428,121,5.524
5,"[17.7, 18.8]",17,0.13,15.712,1.66,0.106,289,18.394
6,"[18.9, 19.5]",10,0.06,6.982,9.109,1.305,100,14.323
7,"[20.0, 21.1]",6,0.03,3.566,5.924,1.661,36,10.095
8,Σ,117,1.00,117.000,—,16.405,—,133.404


4. Докажем, что $χ^2_{набл} = \sum_{i = 1}^k \cfrac{n_i^2}{n_i^{'}} - n$

$χ^2_{набл} = \sum_{i = 1}^k \cfrac{(n_i - n_i^{'})^2}{n_i^{'}}  = \sum_{i = 1}^k \cfrac{n_i^2 - 2n_in_i^{'} + n_i^{'^2}}{n_i^{'}} = \sum \cfrac{n_i^2}{n_i^{'}} - 2\sum n_i + \sum n_i^{'} = \sum \cfrac{n_i^2}{n_i^{'}} - 2n +n = \sum \cfrac{n_i^2}{n_i^{'}} - n$

Проверим для характеристики длина клюва:

$χ^2_{набл} = \sum_{i = 1}^k \cfrac{(n_i - n_i^{'})^2}{n_i^{'}}  = 5,79$

$\sum \cfrac{n_i^2}{n_i^{'}} - n = 122,79 - 117 = 5,79$


Результаты совпали.

Проверим для характеристики высота клюва:

$χ^2_{набл} = \sum_{i = 1}^k \cfrac{(n_i - n_i^{'})^2}{n_i^{'}}  = 16.405$

$\sum \cfrac{n_i^2}{n_i^{'}} - n = 133,405 - 117 = 16,405$

Результаты совпали.




5. По заданному уровню значимости α = 0.05 и числу степеней свободы df найдем критическую точку $χ^2_{крит}$ и сравним с наблюдаемым значением.

Для характеристики длина клюва:

$χ^2_{набл}$ = 5,79

df = k - 3 = 4, тогда

$χ^2_{крит}$ = 9,5

$χ^2_{набл} < χ^2_{крит} -$   гипотеза о нормальном распределении не отвергается, значит нет оснований считать, что характеристика длина клюва распределена не по нормальному закону.

Проверим для высоты клюва:

$χ^2_{набл}$ = 16,405

df = k - 3 = 4, тогда

$χ^2_{крит}$ = 9,5

$χ^2_{набл} > χ^2_{крит} -$   гипотеза о нормальном распределении отвергается, есть основания считать, что характеристика высота клюва распределена не по нормальному закону.

## Выводы

Вычислена точность и доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении при заданном объёме выборки для доверительной точности γ∈{0.95,0.99}. Получено, что при увеличении доверительной вероятности с 0.95 до 0.99 точность оценки уменьшается для обеих характеристик, доверительный интервал становится шире.

Для вычисления границ доверительного интервала для среднеквадратичного отклонения определим значение q при γ∈{0.95,0.99} и n = 117. Построены доверительные интервалы. Для характеристик высота и длина клюва наблюдается закономерность: при увеличении γ доверительный интервал становится шире.

Критерия $χ^2$ (Пирсона) показал, что для характеристики длина клюва гипотеза о нормальном распределении не отвергается, значит нет оснований считать, что характеристика длина клюва распределена не по нормальному закону. Для высоты клюва гипотеза о нормальном распределении отвергается, есть основания считать, что характеристика высота клюва распределена не по нормальному закону.